### PR TITLE
Trap focus within menu pane when tabbing

### PR DESCRIPTION
**Pull request by jhsware**
This only supports tabbing. We don't support:
- key nav in menu pane
- ~~esc to close~~ (FIXED in [c086b5c](https://github.com/KTH/kpm/pull/209/commits/c086b5c47fd438eb9d6f6cd838f808fb91e00ee0))

### CODE REVIEW

**Review by kaj — COMMENTED** [review]
Hm.  Känns lite oroande att implementera tab-to-focus i eget javascript och stänga av browserns hantering i stället för att fixa browserns hantering.

Och hur relaterar det här till förslaget att sätta in den öppna panelen "rätt i ordningen" mellan det valda toplevelalternativet och nästa, som @exacs har föreslagit?

**Comment by exacs** [comment]
I'd say this cannot be released to production until it is possible to close the modal with <kbd>ESC</kbd> key OR a "close button" is actionable inside the modal.

In other words: According to WCAG, users **should** be able to close modals with keyboard-only (either providing a close button, or by pressing ESC, or both).

**Comment by jhsware** [comment]
ESC to close supported with [c086b5c](https://github.com/KTH/kpm/pull/209/commits/c086b5c47fd438eb9d6f6cd838f808fb91e00ee0)

@exacs @kaj when we close the menu pane with ESC, we don't set the focus to the menu item. A user needs to tab from the start of the focus loop. This behaviour is also found in dropdowns, if you have tabbed into the dropdown and press ESC you loose focus. I get that this isn't optimal, but is it acceptable for now?

**Comment by exacs** [comment]
Hmm... Sorry in advance if I sound grumpy!, but [this page in WCAG](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F85#:~:text=Failure%20Example%202,menu%20was%20opened.), describes what is considered to be a failure in the SC 1.4.3 Focus Order:

TL;DR: it is considered a failure if, after dismissing a dialog, the control does not return to the button that opened it _or_ the control does not go to that button after pressing <kbd>Shift + Tab</kbd> (tab back) few times

**Comment by jhsware** [comment]
@exacs I actually appreciate that we are being ambitious on this!

**Comment by jhsware** [comment]
FIXED restoring focus to menu item when closing in 5eeed203f and [727e95f](https://github.com/KTH/kpm/pull/209/commits/727e95f752485d1329c5848a310b71778f630748)
